compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -438,7 +439,16 @@ fn link_rlib<'a>(
     // Add all bundled static native library dependencies.
     // Archives added to the end of .rlib archive, see comment above for the reason.
     for lib in packed_bundled_libs {
-        ab.add_file(&lib)
+        let mut src = fs::File::open(&lib).unwrap();
+        let mut x = Default::default();
+        src.read_to_end(&mut x).unwrap();
```

Comment:
I think you could use `fs::read` here.
Also, an error is actually possible here, so the code should report it instead of unwrapping.